qlib/model/trainer.py

```diff
@@ -441,6 +487,42 @@ def end_train(self, recs, end_train_func=None, experiment_name: str = None, **kw
             before_status=TaskManager.STATUS_PART_DONE,
             **kwargs,
         )
+
+        # FIXME: reset to waiting automatically
+        tm = TaskManager(task_pool=task_pool)
+        for query_task in tm.query({"filter": {"$in": tasks}}):
```

Comment:
Why not use this method TaskManager.waiting?
You may make some improvements to it.